File: hackernews.py
import database
import requests
from bs4 import BeautifulSoup


def scraping_all_pages():
    res = requests.get("https://news.ycombinator.com/news")
    soup = BeautifulSoup(res.text, "html.parser")

    title = soup.select(".storylink")
    vote = soup.select(".subtext")
    news_list = (filtering_hacker_news(title, vote))

    database.initialize_database()
    for news in news_list:
        title = news["Title"].replace("'" ,'"')
        titlelink = news['title_link'].replace("'" ,'"')
        comment = news['comment'].replace("'" ,'"')
        commentlink = news['comments_link'].replace("'" ,'"')
        detail = news['detail'].replace("'" ,'"')
        database.insert_into_hackernews(title,titlelink,comment,commentlink,detail)

    return "hackernews database updated"


def filtering_hacker_news(title, vote):
    news_list = []
    for index, item in enumerate(title):
        url = title[index].get("href")
        point = vote[index].find(class_="score")
        comment = vote[index].find(class_="age")
        comment_link = "https://news.ycombinator.com/" + comment.find('a', href=True).get("href")
        comment = vote[index].getText().split("ago", 1)[1].replace("\xa0", " ").replace("  | hide |", "")
        detail = vote[index].getText().replace("| hide |", "").replace("\n", "").split("ago", 1)[0] + "ago"

        if point:
            current_vote = int(point.getText().replace(" points", ""))
            if current_vote > 99:
                current_news = {'Title': item.getText(), 'title_link': url,'Vote':vote,
                                'comments_link': comment_link, 'detail': detail, 'comment': comment}
                news_list.append(current_news)

    return sorted(news_list[:5], key=lambda k: k['Vote'])

File: frontend/src/components/hackernews/HackerNews.js
import React from "react";
import './HackerNews.css';
import { FcComments } from "react-icons/fc";
import { FcClock } from "react-icons/fc";


const HackerNews = ({ newsList }) => {
    // console.log(newsList);
    return(
        <div className="hackerNews">
            <h2>HackerNews</h2>
           <div>
               <a className="title" href={newsList.titlelink[0]}>{newsList.title[0]}</a>
            </div>
            <div>
                <a className="detail" href="#">{newsList.detail[0].slice(0,-12)}</a><br/>
                <a className="comment mr5" href={newsList.commentlink[0]}><FcComments />{newsList.comment[0].slice(0,-9)}</a><a><FcClock />{newsList.detail[0].slice(-11)}</a>
            </div>
            <hr/>
            <div>
               <a className="title" href={newsList.titlelink[1]}>{newsList.title[1]}</a>
            </div>
            <div>
                <a className="detail" href="#">{newsList.detail[1]}</a><br/>
                <a className="comment mr5" href={newsList.commentlink[1]}><FcComments />{newsList.comment[1].slice(0,-9)}</a><a><FcClock />{newsList.detail[0].slice(-11)}</a>
            </div>
            <hr/>
            <div>
               <a className="title" href={newsList.titlelink[2]}>{newsList.title[2]}</a>
            </div>
            <div>
                <a  className="detail" href="#">{newsList.detail[2]}</a><br/>
                <a className="comment mr5" href={newsList.commentlink[2]}><FcComments />{newsList.comment[2].slice(0,-9)}</a><a><FcClock />{newsList.detail[0].slice(-12)}</a> 
            </div>
            <hr/>
            <div>
               <a className="title" href={newsList.titlelink[3]}>{newsList.title[3]}</a>
            </div>
            <div>
                <a className="detail"  href="#">{newsList.detail[3]}</a><br/>
                <a className="comment mr5" href={newsList.commentlink[3]}><FcComments />{newsList.comment[3].slice(0,-9)}</a><a><FcClock />{newsList.detail[0].slice(0,-12)}</a>
            </div>
            <hr/>
            <div>
               <a  className="title" href={newsList.titlelink[4]}>{newsList.title[4]}</a>
            </div>
            <div>
                <a className="detail" href="#">{newsList.detail[4]}</a><br/>
                <a className="comment mr5" href={newsList.commentlink[4]}><FcComments />{newsList.comment[4].slice(0,-9)}</a><a><FcClock />{newsList.detail[0].slice(-12)}</a>
            </div>
        </div>
    )
}

export default HackerNews;